packages/amplify-cli/src/utils/run-build-command.ts

```diff
@@ -0,0 +1,25 @@
+import { $TSContext, exitOnNextTick, spinner } from 'amplify-cli-core';
+import { sync as execaSync } from 'execa';
+
+export default async function runBuildCommand(context: $TSContext) {
+  if (
+    context.exeInfo.inputParams.runBuildCommand &&
+    context.exeInfo.projectConfig.frontend === 'javascript' &&
+    context.exeInfo.projectConfig?.javascript?.config?.BuildCommand
+  ) {
+    const { BuildCommand } = context.exeInfo.projectConfig.javascript.config;
+    const buildCommandArray = BuildCommand.split(' ');
+    const buildCommandBase = buildCommandArray[0];
+    const buildCommandArgs = buildCommandArray.slice(1);
+    try {
+      spinner.start(`Building app with build command (${BuildCommand}).`);
```

Comment:
I'm not sure it could ever matter, but if someone has a complex custom build command that includes newlines, I'm concerned this spinner could end up printing incorrectly.

I'm thinking it would be simpler if it was broken up. Something like:
```ts
      printer.info(`Executing build command: ${BuildCommand}`);
      spinner.start(`Building app...`);
```